### PR TITLE
Fix #1933 is_numeric check in DB classes.

### LINKED ISSUE #1933: is_numeric check in Mysql update_query
From inc/db_mysql(i).php

				if(is_numeric($value))
				{
					$query .= $comma."`".$field."`={$value}";
				}
				else
				{
					$query .= $comma."`".$field."`={$quote}{$value}{$quote}";
				}

This is only done in `update_query` and not anywhere else (even in Postgres/Sqlite drivers).  
It's problematic anyway, since is_numeric can return true for strings like "0x0", which can cause query errors.

I'm not in a convenient place to check the commit history for why the change was made, but I thought I'd just point it out anyway.

### PR DESCRIPTION
**Pull request by JN-Jones**
Same as #1990 but with the changes I've suggested. @euantorano @Destroy666x

### CODE REVIEW

**Comment by JN-Jones** [comment]
@Destroy666x can we merge this?

File: inc/db_mysql.php [hunk 1]
```diff
@@ -788,7 +788,9 @@ function insert_query($table, $array)
 			}
 			else
 			{
-				$array[$field] = "'{$value}'";
+				$quoted_value = $this->quote_val($value);
+
+				$array[$field] = "{$quoted_value}";
```
Comment (ATofighi) [inline inc/db_mysql.php]: why you don't use:
```php
$array[$field] = $this->quote_val($value);
```
?

Comment (JN-Jones) [inline inc/db_mysql.php]: Guess Euan missed that and simply modified the original line. Will change that (and similar lines)

Comment (JN-Jones) [inline inc/db_mysql.php]: Fixed (hopefully all)

**Comment by PaulBender** [comment]
Works for me. If no one else has any issues, this can be merged.

**Pull request merged**